marklogic-data-hub/src/main/java/com/marklogic/hub/dataservices/ArtifactService.java

```diff
@@ -301,9 +302,11 @@ private ArtifactServiceImpl(DatabaseClient dbClient, JSONWriteHandle servDecl) {
    * Invokes the getList operation on the database server
    *
    * @param artifactType	provides input
+   * @param propertiesToReturn	List of properties to return. Default behavior returns all properties
+   * @param groupByEntityType	If true, artifacts are grouped by Entity Type; default varies by artifact. Artifact types with entity type relationships default to true.
    * @return	as output
    */
-    com.fasterxml.jackson.databind.JsonNode getList(String artifactType);
+    com.fasterxml.jackson.databind.JsonNode getList(String artifactType, Stream<String> propertiesToReturn, Boolean groupByEntityType);
```

Comment:
Rather than changing the getList() method here, will it be easier to add a DS endpoint to StepService (something like getAllSteps.sjs), where we can get all ingestion, mapping (custom in the future) steps and return them ?

Reply:
I agree with that - I think the artifact library is useful for reuse below the service level, but at the service level, life is generally easier with lots of fine-grained endpoints as opposed to fewer coarse-grained ones.

Reply:
Moved to separate DS endpoint.